### ISSUE TITLE
allow OpenAPI enum to be converted to non-enum GQL type

### ISSUE THREAD

**Issue by nlundquist** [IBM/openapi-to-graphql#371]
The enum type in OpenAPI is far more flexible about the possible values than the enum type in GQL. In GQL, enum types are restricted to the [naming convention for symbols in the GQL query language](http://spec.graphql.org/June2018/#sec-Names), while in OpenAPI, the possible values of an enum are essentially limitless.  Although openapi-to-graphql does attempt to sanitize to provide compatibility, there's common scenarios where that sanitization falls short.

Most common would be the removal of compatibility with common standards used in APIs, for example:

```
ISODaysOfTheWeek: {
  type: integer
  enum: 
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
}
```
Each value would have an underscore prefixed to all numbers. 

Another example is an enum of IANA timezone names (eg. `America/Anchorage`, `Africa/Nairobi`), they would all have their slashes removed, leaving users wondering why their queries don't work even though they're following the same standards they used on the corresponding REST API.

Beyond that, `enum`s in JSONSchema aren't limited to scalars, they're applicable to any type, e.g 
```
{ 
  enum: [
    { foo: "bar" },
    { biz: "baz" },
  ]
}
```

One good solution would be to provide a "fragile enums" behavior that would do a check to see if any enum values are inexpressible in the GraphQL enum format, and if so the GQL type resolution will instead fall back to resolving the type as if it was not an enum. (Maybe this should become the default behavior?)

Another possibility would be for JSONSchema types to optionally shortcut the normal openapi-to-graphql type resolution and provide their type GQL type explicitly. For example `DaysOfTheWeek` could include a `x-gql-type-override: "Int"` property.

**Comment by Alan-Cha** [comment]
I agree with your points. Because of the restrictive nature of GraphQL enums, we've always struggled with them. It makes sense to add an option that can allow users to use the original enum values. Previously, users would have to figure out the mapping from original value to the sanitized value, which is not intuitive or easy to use. 

If we were to fall back on the type (if it were not an enum), we can also consider adding the enum values to the description.

I'm not completely convinced we should change the default behavior yet. I need to do some more thinking. One nice thing about the current behavior is that it shows users that it is indeed a feature that we support.

The `x-gql-type-override: "Int"` proposal is certainly interesting. I wonder how useful it is in other cases, however.
